core/templates/services/playthrough.service.ts

```diff
@@ -114,12 +117,15 @@ class CyclicStateTransitionsTracker {
 }
 
 class EarlyQuitTracker {
-  private stateName: string = null;
-  private expDurationInSecs: number = null;
+  // This property is initialized using Angular lifecycle hooks
+  // and we need to do non-null assertion, for more information see
+  // https://github.com/oppia/oppia/wiki/Guide-on-defining-types#ts-7-1
+  private stateName!: string;
+  private expDurationInSecs: number = 0;
```

Comment:
Just to check, why is this not treated similarly to stateName? We shouldn't let this be 0 if that can't occur in practice.

(If it "starts" at 0 and is continuously being incremented, that is a different matter. Just wanted to check which case it is.)

Reply:
OK, thanks for investigating. Could you please explain why 0 is a better choice than null and `!:`? It seems to me that it breaks consistency but I don't understand the rationale behind doing so.

Reply:
Because while using `!:` we are not sure we intialize this with some value before we actually using it (as per above discussion) and null means their is some lack of info, because it is a type of number. Also time can't be negative, their for i use 0.

Also for consistency problem, we also intialize bool variable with false. like i do with 0.

Reply:
Just to be clear, I meant consistency with `stateName`, which isn't initialized. Why is this case different from that?

Also /cc @vojtechjelinek in case he has thoughts on this.

Reply:
Yes i also talking about  consistency with `stateName` . Boolean with false is just an example or approach we do in other parts of codebase. Is consistency is strong reason ?

Reply:
Yes, we should have one approach everywhere in the codebase, ideally. But please check with @vojtechjelinek on what that "default approach" should be.

Reply:
Hmm, I think we should not use zero here. The reason is consistency with `stateName`.

I think the case when we would use zero is if were adding seconds to `expDurationInSecs`. So if we had something like `addDuration(x) {self.expDurationInSecs += x}` as in that case the `expDurationInSecs` clearly needs to have some starting value. In this case, we always set `expDurationInSecs` and thus there is (very improbable) possibility of `expDurationInSecs` being set to zero and then we wouldn't be able to spot the difference.

Reply:
Done